### DISCUSSION TITLE
[SDK] import * as GREETER_CONTRACT_ABI from "./abi.json"; // TODO: Complete and import the ABI

### DISCUSSION THREAD

**Discussion by KK-yao** [zkSync-Community-Hub/zksync-developers#374] [SDKs] [answered]
### Environment

Testnet

### Acknowledgement

- [ ] I understand that issues are expected with the Go SDK as they are being updated.

### Issue Description

import * as GREETER_CONTRACT_ABI from "./abi.json"; // TODO: Complete and import the ABI

const initializeProviderAndSigner = async () => {
  provider = new Provider("https://sepolia.era.zksync.dev");
  // Note that we still need to get the Metamask signer
  signer = await new BrowserProvider(window.ethereum).getSigner();
  contract = new Contract(GREETER_CONTRACT_ADDRESS, GREETER_CONTRACT_ABI, signer);
};
new Contract() reported an error：
TypeError: abi is not iterable               zksync-ethers.js?v=ac7c7726:13786
I think there is a problem reading ./abi.json。
I changed it to the following code and it passed:

const initializeProviderAndSigner = async () => {
  provider = new Provider("https://sepolia.era.zksync.dev");
  // Note that we still need to get the Metamask signer
  signer = await new BrowserProvider(window.ethereum).getSigner();

  const erc_abi = [
    "function greet() view returns(string)"
];

  contract = new Contract(GREETER_CONTRACT_ADDRESS, erc_abi, signer);
};

### Expected Behavior

none

### Code Example

none

**Comment by therntibmaph** [comment]
It appears that there was an issue with importing the ABI from the JSON file. Ensure that the ABI is properly formatted and exported in the JSON file. Alternatively, define the ABI directly in the code as shown in the provided workaround.

**Comment by uF4No** [comment, marked as answer]
The `abi.json` file from the [frontend quickstart tutorial](https://docs.zksync.io/build/tutorials/dapp-development/frontend-quickstart-paymaster.html) is not the full artifact generated by the compiler, but only the `abi` part of it. That's why it worked when you created the `erc_abi` variable. You can either do that or, if you copy/past the full artifact, reference it using `GREETER_CONTRACT_ABI.abi` later on.

**Comment by mattbrathwaite** [comment]
It seems that there might have been an issue with the format or structure of the ABI imported from the JSON file. By manually defining the ABI in code, the issue was resolved. Ensure that the ABI JSON file is correctly formatted and compatible with the way it's being imported.

**Comment by propmacboorsp** [comment]
It seems there was an issue with importing the ABI from "./abi.json" file, resulting in a TypeError when initializing the contract. The problem was resolved by directly defining the ABI in the code. This suggests a potential problem with reading the ABI file or its format. Further investigation may be needed to ensure proper handling of the ABI file or to troubleshoot any issues with its content or import process.

**Comment by erinemsyntracy** [comment]
The user encountered an error when trying to use the new Contract() function with an ABI imported from a JSON file. The error message was TypeError: abi is not iterable. The user was able to resolve the issue by defining the ABI as an array directly in the code instead of importing it from a JSON file. Without further information or access to the code repository, it is difficult to diagnose the exact issue.
